src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -145,6 +144,12 @@ public void run(@NotNull ProgressIndicator progressIndicator) {
     }
 
     public static void setupCodeSyncRepo(Project project, String repoPath, String repoName, CodeSyncProgressIndicator codeSyncProgressIndicator, boolean skipSyncPrompt, boolean isSyncingBranch) {
+
+        if(Utils.isIndividualFileOpen(project)){
```

Comment:
When will code reach here? I mean "Connect" option is disabled for singe-file-opened-as-project, right?

Reply:
This method is being called on startup and it will also produce error message to user when they try to click "Connect" button.

Reply:
Point is, why would user even see the "Connect" button? It should not be visible, right?

Reply:
@syedgulahmed We can disable the connect button from `CodeSyncSetupAction` here https://github.com/codesyncapp/codesync-intellij/blob/master/src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetupAction.java#L57 is an example.

Reply:
Well `setupCodeSyncRepo` is called in `setupCodeSyncRepoAsync` method that method has calls in in OpenCloseListener  and PopulateBuffer.